module/VuFind/src/VuFind/Cache/KeyGeneratorTrait.php

```diff
@@ -58,7 +58,8 @@ protected function getCacheKey($suffix = null)
         // Test the build key
         if (
             $this->cache
-            && !preg_match($this->cache->getOptions()->getKeyPattern(), $key)
+            && ($keyPattern = $this->cache->getOptions()->getKeyPattern())
```

Comment:
Is this a bug fix worth merging separately?

Reply:
Yes, I'll open a PR.

Reply:
PR #2892.

Reply:
Thanks! I've merged that back into this PR to reduce the file count.